libshaderc/src/shaderc_test.cc

```diff
@@ -482,6 +483,149 @@ TEST_F(CompileStringWithOptionsTest, PreprocessingOnlyOption) {
   EXPECT_THAT(preprocessed_text_cloned_options, HasSubstr("void main(){ }"));
 }
 
+// To test file inclusion, use an unordered_map as a fake file system to store
+// fake files to be included. The unordered_map represents a filesystem by
+// mapping filename string (or path) to the contents of that file as a string.
+using FakeFS = std::unordered_map<std::string, std::string>;
+
+// An includer test case needs: 1) A fake file system which is actually an
+// unordered_map, so that we can resolve the content given a string. A valid
+// fake file system must have one entry with key:'root' to specify the start
+// shader file or compilation.2) An string that we expect to see in the
+// compilation output.
+class IncluderTestCase {
+ public:
+  IncluderTestCase(FakeFS fake_fs, std::string expected_substring)
+      : fake_fs_(fake_fs), expected_substring_(expected_substring) {
+    assert(fake_fs_.find("root") != fake_fs_.end() &&
+           "Valid fake file system needs a 'root' file\n");
+      };
+
+  const FakeFS& fake_fs() const { return fake_fs_; }
+  const std::string& expected_substring() const { return expected_substring_; }
+
+ private:
+  FakeFS fake_fs_;
+  std::string expected_substring_;
+};
+
+// A mock class that simulate an includer. C API needs two function pointers
+// each for get including data and release the data. This class defined two
+// static functions, which wrap their matching member functions, to be passed to
+// libshaderc C API.
+class TestIncluder {
+ public:
+  TestIncluder(const FakeFS& fake_fs) : fake_fs_(fake_fs), responses_({}){};
+
+  // Get path and content from the fake file system.
+  shaderc_includer_response* GetInclude(const char* filename) {
+    responses_.emplace_back(shaderc_includer_response{
+        filename, strlen(filename), fake_fs_.at(std::string(filename)).c_str(),
+        fake_fs_.at(std::string(filename)).size()});
+    return &responses_.back();
+  }
+
+  // Response data is owned as private property, no need to release explicitly.
+  void ReleaseInclude(shaderc_includer_response* data) { (void)data; };
```

Comment:
just drop the parameter name.  Then you can just write this:
```
void ReleaseInclude(shaderc_includer_response*) { }
```

(Also, the trailing semicolon is redundant and should be removed.)